File: backend/app/Models/SummaryDiscussion.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class SummaryDiscussion extends Model
{
    use HasFactory;

    protected $table = 'summary_discussions';
    protected $primaryKey = 'id_summarydiscussion';
    public $incrementing = true;
    protected $keyType = 'int';

    protected $fillable = [
        'content',
        'fk_id_user',
        'fk_id_chatroomai',
    ];
}

File: sea_app/lib/models/summary_discussion.dart
class SummaryDiscussion {
  final String id;
  final String content;
  final String? fkIdUser;
  final String? fkIdChatroomAi;

  SummaryDiscussion({required this.id, required this.content, this.fkIdUser, this.fkIdChatroomAi});

  static SummaryDiscussion fromJson(Map<String, dynamic> j) {
    return SummaryDiscussion(
      id: (j['id_summarydiscussion'] ?? j['id'] ?? '').toString(),
      content: (j['content'] ?? '').toString(),
      fkIdUser: (j['fk_id_user'] ?? j['fk_id_user'])?.toString(),
      fkIdChatroomAi: (j['fk_id_chatroomai'] ?? j['fk_id_chatroomai'])?.toString(),
    );
  }
}
